integration/query_evaluation_test.go

```diff
@@ -464,15 +482,25 @@ func TestQueryEvaluationRegex(t *testing.T) {
 			expectedIDs: []any{},
 		},
 		"RegexBadOption": {
-			filter:      bson.D{{"v", bson.D{{"$regex", primitive.Regex{Pattern: "foo", Options: "123"}}}}},
-			expectedIDs: []any{"multiline-string", "string"},
+			filter: bson.D{{"v", bson.D{{"$regex", primitive.Regex{Pattern: "foo", Options: "123"}}}}},
+			err: &mongo.CommandError{
+				Code:    51108,
+				Name:    "Location51108",
+				Message: ` invalid flag in regex options: 1`,
+			},
+			altMessage: `invalid flag in regex options: 1`,
```

Comment:
I _think_ altMessage was supposed to be MongoDB's error, not vise versa.

Let's rename `AssertEqualAltError` argument names and update docs to mention that if that's true